core/lib/Cardano/Address/Style/Shelley.hs

```diff
@@ -490,25 +495,24 @@ inspectShelleyAddress addr
                         ]
                -- 0110: enterprise address: keyhash28
                 0b01100000 | BS.length rest == pubkeyHashSize ->
-                    Just $ Json.object
+                    pure $ Json.object
                         [ "address_style"     .= Json.String "Shelley"
                         , "stake_reference"   .= Json.String "none"
                         , "spending_key_hash" .= base16 (BS.take pubkeyHashSize rest)
                         , "network_tag"       .= network
                         ]
                -- 0111: enterprise address: scripthash32
                 0b01110000 | BS.length rest == scriptHashSize ->
-                    Just $ Json.object
+                    pure $ Json.object
                         [ "address_style"     .= Json.String "Shelley"
                         , "stake_reference"   .= Json.String "none"
                         , "script_hash"       .= base16 (BS.take scriptHashSize rest)
                         , "network_tag"       .= network
                         ]
                -- 1000: byron address
                 0b10000000 ->
-                    inspectByronAddress addr <|> inspectIcarusAddress addr
-                _ ->
-                    Nothing
+                  inspectByronAddress addr <|> inspectIcarusAddress addr
```

Comment:
I wonder how this plays in plain IO with regard to the `MonadThrow` constraint. I didn't even know there was an `Alternative` instance for IO and at this stage, I am a bit puzzled about which intuition I'd have for it :thinking:

Reply:
MonadThrow + IO always ends up es exceptions. Alternative in IO will then catch them and try the other branch.

Reply:
Sweet.

Reply:
Actually, not true :cry: ... MonadThrow + IO results in Exceptions, but these don't specialize to `IOException`... and `(<|>)` only catches those.